server/middlewares/errorHandler.js

```diff
@@ -0,0 +1,35 @@
+const {
+    NotFoundError,
+    UnauthorizedError,
+    AuthenticationError,
+    UnavailableServiceError,
+    BadRequestError } = require('../errorTypes/errorTypes')
+
+module.exports = handleError = (req, res, next) => {
+    res.handleError = (err, res)=>{
+        switch (err.constructor) {
+            case NotFoundError:
+                return res.status(404).json({
+                    message:err.message
+                })
+            case UnauthorizedError:
+                return res.status(401).json({
+                    message:err.message
+                })
+            case AuthenticationError:
+                console.log('auth failed')
+                return res.status(403).json({
+                    message:err.message
```

Comment:
I think it is better to have some predefined messages
message: err.message || 'Authentication failed'

Reply:
Do you mean messages stored in some file like:
`const AUTHENTICATION_ERROR = 'Authentication failed'`
or just messages displayed when err.message is not defined like in your example?

Added default message to every type of error.